pypesto/store/read_from_hdf5.py

```diff
@@ -294,21 +294,40 @@ def read_result(filename: str,
 
     if problem:
         pypesto_problem_reader = ProblemHDF5Reader(filename)
-        result.problem = pypesto_problem_reader.read()
+        try:
+            result.problem = pypesto_problem_reader.read()
+        except KeyError:
+            logger.warning('Loading the problem failed. It is'
+                           'highly likely that no problem exists.')
```

Comment:
```suggestion
            logger.warning('Loading the problem failed. It is'
                           'highly likely that no problem exists '
                           f'within {filename}.')
```

To match the other messages.

Reply:
As Problem should always be written and read I corrected this to a regular warning that loading the problem failed. As of now I think most problems have amici objectives which have to be created again so that having the problem read in fail is in most cases no hindrance to the code.

Reply:
Hm then it would be handled differently to sample/optimize/profile results. There, only `KeyError` is handled to treat a known issue, and other exceptions would be raised in the usual way. Here, all exceptions are caught and replaced with a `logger.warning`. So it may be better/more consistent to remove the try-except for the `pypesto.Problem`.

If there is a particular exception in mind here, better to use the specific constant, (e.g. `KeyError` instead of `Exception`).

Just a small point, fine to ignore.

Reply:
Think you are right, just removed it as there is no particular exception👍